app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/minigame/AsteroidImpactActivity.java

```diff
@@ -61,16 +64,17 @@ public void createProgress(View view) {
      * @param progressBar the progressBar on which progress must increase.
      */
     public void increaseProgress(ProgressBar progressBar) {
+        progressBar.incrementProgressBy(5);
+
         if (progressBar.getProgress() >= 95) {
             EventCompletedMessage message = new EventCompletedMessage(EventTypes.ASTEROID_IMPACT.ordinal());
             Map<String, Integer> parameters = new HashMap<String, Integer>();
             parameters.put("locationDamageImpact", (repairingEnergyShield ? 0 : 1));
             message.setParameters(parameters);
             game.sendMessage(message);
-        }
 
-        if (progressBar.getProgress() != 100) {
-            progressBar.incrementProgressBy(5);
```

Comment:
I'm not sure why people touched this, because now you can keep caling the method to increment, and I did it in this order (first checking for 95, to achieve consistency).